hist/hist/inc/THnBase.h

```diff
@@ -63,6 +65,10 @@ class THnBase: public TNamed {
     THnBase(const char *name, const char *title, Int_t dim, const Int_t *nbins, const Double_t *xmin,
             const Double_t *xmax);
 
+    THnBase(const char* name, const char* title, const std::vector<TAxis*>& axes);
+
+    THnBase(const char* name, const char* title, const std::vector<TAxis>& axes);
```

Comment:
The THnBase is an internal base class. It's not necessary to have the convenient constructor for the initializer lists here. I think the `const std::vector<TAxis*>&` version is enough.